script/stresstest.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+#
+# stresstest.sh is a helper script to run specific Go tests with x/tools/cmd/stress.
+#
+# Usage: ./script/stresstest.sh [package_path] ^TestName$ [stress options...]
+#
+# Example: ./script/stresstest.sh ./libbeat/common/backoff ^TestBackoff$ -p 32
+
+set -e
+
+test_package_path=${1:-$(pwd)}
```

Comment:
`${1:-$(pwd)}` suggests that this argument is optional and points to the current directory by default. However, since you have more arguments in this script, it's not possible. If you don't specify the path `^TestBackoff$` becomes `$1`.

I think it would be nice to find a way to build this tool so you don't have to specify the path every time you run it.

Can we simplify it to the name of the test (not a regexp) and the amount of iterations? I think it's okay to document that you need to run this script from the package you want to test and expect the user to change the directory.

P.S. I don't think it should be a mage target.

Reply:
I like having it be a regexp, name of the test can get a little awkward for some of the cases where long names are spliced together, and it's nice that it also covers running a particular test suite and not just a single test (I'd like the ability to require an exact match _or_ to give just a substring, and regex is the most obvious way to do both in 10 lines).

I think it's reasonable to require the directory as an argument, though I'd like if it quit with an error if there aren't enough arguments.

I also don't think it should be a mage target, the mage build is 5K+ lines of complex semi-documented monolithic code and I don't think a 10-line shell script would gain anything by being merged into it.

Reply:
I like the regex approach because of it's versatility, it is the same as `-test.run`. I updated the code to validate that the path exists and also added a --help option.